awsiam/iam_user.go

```diff
@@ -27,6 +27,25 @@ func NewIAMUser(
 	}
 }
 
+func (i *IAMUser) Exists(userName string) (bool, error) {
+	existsUserInput := &iam.GetUserInput{
+		UserName: aws.String(userName),
+	}
+	i.logger.Debug("exists-user", lager.Data{"input": existsUserInput})
+	_, err := i.iamsvc.GetUser(existsUserInput)
+	if err != nil {
+		i.logger.Error("exists-user.aws-iam-error", err)
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "NoSuchEntity" {
+				return false, nil
+			}
+		}
+		// FIXME: returning false seems wrong here
+		return false, err
```

Comment:
Nope, it's not wrong. In Go, you always want the return value to match the function signature, which is `(bool, error)`. Even though it feels like you're saying something definitive by returning `false` when there is really an error, it is idiomatic in Go to always check the error for being non-nil when it is returned.